### ISSUE TITLE
bug report: relay map does not take effect

### ISSUE THREAD

**Issue by gszr** [docker-mailserver/docker-mailserver#3954]
### 📝 Preliminary Checks

- [X] I tried searching for an existing issue and followed the [debugging docs](https://docker-mailserver.github.io/docker-mailserver/latest/config/debugging/) advice, but still need assistance.

### 👀 What Happened?

Howdy, all!

I added a relay for an alias I have, like so:
```
./setup.sh relay add-domain acm.org smtpauth.mailroute.net 587
./setup.sh relay add-auth acm.org gsz@acm.org <password>
```

Verified that the configs (mail/config/postfix-relaymap.cf and mail/config/postfix-relaymap.cf) are indeed mapped in the container - plus, tried a down & up to restart.

However, when I send an email (from my web client, Rainloop, where I configured the alias as an identity), the relay does not take effect; I see the following in the logs:
```
Mar 25 02:19:45 mail postfix/qmgr[1641]: 461712FBA41: from=<gsz@acm.org>, size=538, nrcpt=1 (queue active)
Mar 25 02:19:45 mail amavis[919]: (00919-02) Passed CLEAN {RelayedOpenRelay}, [172.20.0.1]:53156 <gsz@acm.org> -> <gsz@fastmail.com>, Queue-ID: 0D4BB2F93E1, Message-ID: <c9c994288e53afed2466d89a70391bc7@acm.org>, mail_id: 7WjR_ZgIYiMg, Hits: -0.002, size: 313, queued_as: 461712FBA41, 1178 ms
Mar 25 02:19:45 mail postfix/smtp-amavis/smtp[2509]: 0D4BB2F93E1: to=<gsz@fastmail.com>, relay=127.0.0.1[127.0.0.1]:10024, delay=1.3, delays=0.06/0.01/0/1.2, dsn=2.0.0, status=sent (250 2.0.0 from MTA(smtp:[127.0.0.1]:10025): 250 2.0.0 Ok: queued as 461712FBA41)
Mar 25 02:19:45 mail postfix/qmgr[1641]: 0D4BB2F93E1: removed
```

That is, relay shows up as `relay=127.0.0.1[127.0.0.1]:10024`.

Running on version `13.3.1`.

### 🐋 DMS Version

13.3.1

### 💻 Operating System and Architecture

Debian 12, arm64

### ⚙️ Container configuration files

```yml
mail:
    image: mailserver/docker-mailserver:13.3.1
    restart: always
    hostname: mail
    domainname: <my domain>
    container_name: mail
    ports:
      - "25:25"
      - "465:465"
      - "587:587"
      - "993:993"
      - "4190:4190"
      - "11334:11334"
    volumes:
      - mail-data:/var/mail:Z
      - mail-state:/var/mail-state:Z
      - ./mail/config:/tmp/docker-mailserver/
    environment:
      - ENABLE_SPAMASSASSIN=1
      - ENABLE_CLAMAV=1
      - ENABLE_FAIL2BAN=1
      - ENABLE_POSTGREY=1
      - ONE_DIR=1
      - DMS_DEBUG=1
      - ENABLE_MANAGESIEVE=1
      - SSL_TYPE=manual
      - SSL_CERT_PATH=/run/secrets/ssl_cert
      - SSL_KEY_PATH=/run/secrets/ssl_cert_key
      - POSTFIX_MESSAGE_SIZE_LIMIT=51200000
      - SPAMASSASSIN_SPAM_TO_INBOX=1
      - MOVE_SPAM_TO_JUNK=1
      - SA_KILL=30
      - POSTGREY_AUTO_WHITELIST_CLIENTS=3
    cap_add:
      - NET_ADMIN
    secrets:
      - ssl_cert
      - ssl_cert_key
```

**Comment by gszr** [comment]
Folks, I might have been a bit hasty on opening this as an issue. I will take some more time to investigate what is going on and reopen if needed. Apologies for the noise.

**Comment by gszr** [comment]
In case others come across this issue.
I was trying to route only emails sent from my @acm.org alias through ACM's Mailroute smtp server. I did run both `./setup.sh relay add-domain` and `./setup.sh relay add-auth acm.org`, but it still did not work. Looking at [the code](https://github.com/docker-mailserver/docker-mailserver//blob/0c7e49e654fb64a0b23abae2b423f54e8312beeb/target/scripts/helpers/relay.sh#L190), I saw that `RELAY_HOST` needs to be set for the relay map to apply. I also found https://github.com/docker-mailserver/docker-mailserver/issues/3607, which seems to be fixed (I did not test) in the master branch. Setting `RELAY_HOST` to " " - just to test - the acm.org map worked, but now any mail sent from my smtp server fails.
